doc/Customization-guide/Endpoint-SDKs/C-SDK/Porting-guide/index.md

```diff
@@ -33,34 +33,27 @@ If you want to port it, you need to create a few files of the SDK and implement
  - Output stream, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/stdio.h).
  
  - Default settings, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/defaults.h).
+ 
+ - SHA-1 calculation (you may use a default implementation), [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform/ext_sha.h).  
   
 > **NOTE:** All references are files which are used by esp8266. You need to use the same but with own implementations and put ones to a folder which needs to create in the [platform-impl]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/). The new folder needs to have the same folder tree as esp8266 folder. 
  
 ## Optional         
 
- - Encryption. Compile-time generated and run-time generated keys of the encryption. For example, the ESP8266 demo uses only the compile-time generated key.
-
- - SHA-1 calculation (you may use a default implementation).
- 
- - Log and debug.  
+ - Encryption. Compile-time generated and run-time generated encryption keys. For example, ESP8266 demos use only the compile-time generated key.
  
-> **NOTE:** Consider to use the compile-time generated key in case of constrained, bare-metal platforms to save MCU resources and the run-time generated key doesn't give a big dispersion  on small platforms.
+ - Debugging log.  
```

Comment:
I would say this is a minor nitpick.
@rasendubi , @forGGe are there any concerns from the technical point of view? If not, let accept this PR. It will be reviewed by a tech writer anyway. 
I am Ok with the deep review, however, it seems like an endless story.